### PR TITLE
feat(ElasticSearchVectorStore): Added another es vs that uses regular…

### PR DESCRIPTION
**Pull request by MFarrasHakim**
… NewClient instead of TypedClient

When using the es from staging, using typed api caused an error Invalid media-type value on headers [Accept, Content-Type] when sending requests to the es server.
So added an option to send requests to es server without using typed api

### CODE REVIEW

**Comment by wejick** [comment]
Please add unit test to `dataToESDoc` and `hitToDocs`

File: datastore/reg_elasticsearch_vector/reg_elasticsearch.go [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+// AddText store text to vector index
+// it will also store embedding of the text using specified embedding model
+// If the index doesnt exist, it will create one with default schema
+func (ES *ElasticsearchVectorStore) AddText(ctx context.Context, indexName string, input string) (err error) {
+	_, err = ES.AddDocuments(ctx, indexName, []document.Document{{Text: input}})
+	return
+}
+
+// A function for marshaling structs to JSON string
+func jsonStruct(doc interface{}) (string, error) {
```
Comment (wejick) [inline datastore/reg_elasticsearch_vector/reg_elasticsearch.go:115]: add unit test

File: datastore/reg_elasticsearch_vector/reg_elasticsearch.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+	return
+}
+
+// DeleteIndex drop the index
+func (ES *ElasticsearchVectorStore) DeleteIndex(ctx context.Context, indexName string) (err error) {
+	_, err = ES.regularEsClient.Indices.Delete(
+		[]string{indexName},
+		ES.regularEsClient.Indices.Delete.WithContext(ctx),
+	)
+	// _, err = ES.esClient.API.Indices.Delete(indexName).Do(ctx)
```
Comment (wejick) [inline datastore/reg_elasticsearch_vector/reg_elasticsearch.go:176]: remove

File: datastore/reg_elasticsearch_vector/reg_elasticsearch.go [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+			NumCandidates: 5,
+		},
+		Fields: []string{"dense_vector"},
+	}
+
+	jsonVector, err := jsonStruct(knnSearchBody)
+	if err != nil {
+		return
+	}
+	// log.Println("============ jsonVector =================")
```
Comment (wejick) [inline datastore/reg_elasticsearch_vector/reg_elasticsearch.go:213]: clean up

File: datastore/reg_elasticsearch_vector/reg_elasticsearch.go [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+		ES.regularEsClient.API.Indices.Exists.WithContext(ctx),
+	)
+	if err != nil {
+		return false, err
+	}
+	if resp.StatusCode == 400 || resp.StatusCode == 404 {
+		return false, nil
+	}
+	exist = true
+	// exist, err = ES.esClient.API.Indices.Exists(indexName).IsSuccess(ctx)
```
Comment (wejick) [inline datastore/reg_elasticsearch_vector/reg_elasticsearch.go:267]: clean up

File: datastore/reg_elasticsearch_vector/reg_elasticsearch.go (hunk 1 above)
Comment (MFarrasHakim) [reply datastore/reg_elasticsearch_vector/reg_elasticsearch.go]: updated ya pak. Please re review @wejick

File: datastore/elasticsearch_vector/elasticsearch_test.go [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+		doc interface{}
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "Success",
```
Comment (wejick) [inline datastore/elasticsearch_vector/elasticsearch_test.go:21]: add test case for empty data

File: datastore/elasticsearch_vector/elasticsearch_test.go [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+		esRespBody       ElasticResponse
+		additionalFields []string
+	}
+	tests := []struct {
+		name     string
+		args     args
+		wantDocs []document.Document
+		wantErr  bool
+	}{
+		{
```
Comment (wejick) [inline datastore/elasticsearch_vector/elasticsearch_test.go:62]: add test case for empty data

File: datastore/elasticsearch_vector/elasticsearch_test.go [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+		documents []document.Document
+		vector    [][]float32
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantOutput []elasticDocument
+	}{
+		{
+			name: "success",
```
Comment (wejick) [inline datastore/elasticsearch_vector/elasticsearch_test.go:155]: add test case for empty data

**Review by wejick — APPROVED** [review]
